lib/dpl/provider/elastic_beanstalk.rb

```diff
@@ -39,7 +39,11 @@ def push_app
         create_bucket unless bucket_exists?
 
         if options[:zip_file]
-          zip_file = File.join(Dir.pwd, options[:zip_file])
+          if File.dirname(options[:zip_file]) == "."
```

Comment:
Using the `expand_path` method without the `dir_string` creates an `absolute_path` then probably we don't need the `absolute_path` . We could use the `absolute_path` for situations where we want to maintain our path in the current scope(project directory) without expand the path to the root/home folders in the case we should remove the expand_path method or we could use one of them and set the `dir_string` 

```
# assuming $HOME env var is not empty

BUILD_DIR = '~/project_directory/' 
options[:zip_file] =  "#{BUILD_DIR}/artifact.zip"
zip_file = File.expand_path(options[:zip_file]) # -> /home/user/project_directory/artifact.zip

BUILD_DIR = '~other/project_directory/' 
options[:zip_file] =  "#{BUILD_DIR}/artifact.zip"
zip_file = File.expand_path(options[:zip_file]) # -> /home/other/project_directory/artifact.zip

```

And we could use `absolute_path` when we want to maintain the zip_file in the same home directory

```
  BUILD_DIR = '~other/project_directory/' 
  options[:zip_file] =  "#{BUILD_DIR}/artifact.zip"
  zip_file = File.expand_path(options[:zip_file]) # -> /home/root/~other/project_directory/artifact.zip

  BUILD_DIR = '~/project_directory/' 
  options[:zip_file] =  "#{BUILD_DIR}/artifact.zip"
  zip_file = File.expand_path(options[:zip_file]) # -> /home/root/~/project_directory/artifact.zip
```